### PR TITLE
Persist runner request order

### PR DESCRIPTION
**Pull request by jackkav**
- persist runner request order
- fix key value sort order (would previously only let you move up or down one position)
- remove useListData

future work
- moving multiple items reverses the order

Ref: INS-4452
Ref #7980

### CODE REVIEW

**Review by filfreire — DISMISSED** [review]

**Comment by ihexxa** [comment]
I propose to take care of at least following 2 problems before moving forward:
- User could change requests after settings (request selection) are persisted.
- The persisted settings should be saved for different workspaces, as requests don't not exist for another workspace.

And the approach of persisting selection looks good.

File: packages/insomnia/src/ui/routes/runner.tsx [hunk 1]
```diff
@@ -215,6 +215,15 @@ export const Runner: FC<{}> = () => {
       };
     });
   const [allKeys, setAllKeys] = useLocalStorage<string[]>(localStorageKey + 'allKeys', { defaultValue: requestRows.map(item => item.id) });
+  // request was added or removed
+  if (allKeys.length !== requestRows.length) {
```
Comment (ihexxa) [inline packages/insomnia/src/ui/routes/runner.tsx]: I think the condition is more complicated, e.g. this check might not work in this case: user adds one new request and deletes one while actually one request has changed.

File: packages/insomnia/src/ui/routes/runner.tsx [hunk 2]
```diff
@@ -215,6 +215,15 @@ export const Runner: FC<{}> = () => {
...
+    const newRequest = requestRows.find(item => !allKeys.includes(item.id));
+    if (newRequest) {
+      setAllKeys([...allKeys, newRequest.id]);
+    } else {
+      setAllKeys(allKeys.filter(key => requestRows.map(r => r.id).includes(key)));
```
Comment (ihexxa) [inline packages/insomnia/src/ui/routes/runner.tsx]: The user could move one request to another place.

Comment (jackkav) [reply packages/insomnia/src/ui/routes/runner.tsx]: thats true, feel free to create a new PR to address this.

**Pull request merged**